SoftLayer/managers/vs.py

```diff
@@ -368,10 +368,25 @@ def _generate_create_dict(
             data.update({
                 'primaryNetworkComponent': {
                     "networkVlan": {"id": int(public_vlan)}}})
+
         if private_vlan:
-            data.update({
-                "primaryBackendNetworkComponent": {
-                    "networkVlan": {"id": int(private_vlan)}}})
+            # As per https://stackoverflow.com/questions/37592080/create-a-softlayer-virtual-guest-on-a-specific-subnet
+            # we can specify a subnet under the VLAN
+            if private_vlan_subnet:
```

Comment:
This shouldn't be inside the private_vlan if statement since I don't think the API requires a vlan id when supplying a subnet id.